### PR TITLE
feat: Improve page type and parameters detection

### PR DESCRIPTION
**Pull request by igorbrasileiro**

### CODE REVIEW

File: components/VTEXPortalDataLayerCompatibility.tsx [hunk 1]
```diff
@@ -15,7 +15,33 @@ function addVTEXPortalDataSnippet(accountName: any) {
...
+  const breadcrumbSD = structuredDatas.find((
+    s,
+  ) => (s["@type"] === "BreadcrumbList"));
+
+  // deno-lint-ignore no-explicit-any
+  const getPageType = (hasStructuredData: undefined | Record<string, any>) => {
+    if (url.pathname === "/") return "homeView";
+
+    const isProductPage = structuredDatas.some((s) => s["@type"] === "Product");
+    if (isProductPage) return "productView";
```
Comment (lucis) [inline components/VTEXPortalDataLayerCompatibility.tsx:27]: This doesn't get triggered by pages that have products inside of them?

**Review by lucis — APPROVED** [review]
Nice!

File: components/VTEXPortalDataLayerCompatibility.tsx [hunk 2]
```diff
@@ -15,7 +15,33 @@ function addVTEXPortalDataSnippet(accountName: any) {
...
+
+    const isSearchPage = url.pathname.startsWith("/s");
```
Comment (tlgimenes) [inline components/VTEXPortalDataLayerCompatibility.tsx:29]: e o caso de `/sacolas` ?

Comment (igorbrasileiro) [reply components/VTEXPortalDataLayerCompatibility.tsx:29]: Booa noobei

File: components/VTEXPortalDataLayerCompatibility.tsx (hunk 1 above)
Comment (igorbrasileiro) [reply components/VTEXPortalDataLayerCompatibility.tsx:27]: Yeah, IMO does it make sense. If a page that represent a product page and doesn't have a product ld+json, something is wrong.

**Pull request merged**